Userland/Libraries/LibC/mallocdefs.h

```diff
@@ -19,6 +19,14 @@
 static constexpr unsigned short size_classes[] = { 16, 32, 64, 128, 256, 496, 1008, 2032, 4080, 8176, 16368, 32752, 0 };
 static constexpr size_t num_size_classes = (sizeof(size_classes) / sizeof(unsigned short)) - 1;
 
+extern "C" {
+extern
+#ifndef NO_TLS
```

Comment:
Shouldn't we error out / disable the feature if there is no tls? As this feature is essentially broken?

Reply:
So I get that all of the allocation prevention stuff should live behind the #ifndef NO_TLS ?

Reply:
For now, the DynamicLoader just doesn't have the VERIFY's, making all NoAllocationGuard-stuff a no-op.